specifyweb/frontend/js_src/lib/components/TreeView/Actions.tsx

```diff
@@ -76,6 +76,8 @@ export function TreeViewActions<SCHEMA extends AnyTree>({
   const disableButtons =
     focusedRow === undefined || typeof currentAction === 'string';
 
+  const isViewMode = !hasTablePermission(tableName, 'update');
+
```

Comment:
Just to make things consistent with other parts of Specify, I wonder if we could use `ReadonlyContext` here?
We can do something like
```
const isReadOnly = React.useContext(ReadOnlyContext);
```
and then use `isReadOnly` instead of `isViewMode`.

Functionally nothing will change but it will just ensure we handle readonly mode in the same way everywhere.